From 5b9df34e09da9091d19355fba8236d3c36e24660 Mon Sep 17 00:00:00 2001
From: baizg1107 <preloyalwhite@163.com>
Date: Fri, 13 May 2022 10:46:46 +0800
Subject: [PATCH] bird 2.0.9 babel iface

---
 proto/babel/babel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/proto/babel/babel.c b/proto/babel/babel.c
index e43818f..174fc9e 100644
--- a/proto/babel/babel.c
+++ b/proto/babel/babel.c
@@ -1899,7 +1899,7 @@ babel_reconfigure_ifaces(struct babel_proto *p, struct babel_config *cf)
     struct babel_iface *ifa = babel_find_iface(p, iface);
     struct babel_iface_config *ic = (void *) iface_patt_find(&cf->iface_list, iface, NULL);
 
-    if (ic && iface_is_valid(p, iface))
+    if (ic && !iface_is_valid(p, iface))
       ic = NULL;
 
     if (ifa && ic)
-- 
2.27.0

